### PR TITLE
Prevents some duplicate calls of update_icon() during explosions

### PR DESCRIPTION
**Pull request by Chewyfungus**
### update_icon() is needlessly called many times during explosions. ###
----
* A pipe that's destroyed will call update_icon() on itself and on any nearby pipes that will be visually affected by this pipe disappearing. 
* It's a pretty meaty proc, so this isn't a very good thing to have happen when it doesn't need to.
----
### The solution that clusterfack suggested I implement was the following:
* Whenever a pipe would call update_icon(), instead add it to a list in its parent pipeline, and set a flag equal to true
* In the parent pipeline, whenever something is added to this list, the list begins processing this as a queue by taking the pipe stored in the list and calling update_icon() for it if its update flag is set to true. 
* Then, after update_icon() is called, set this flag to false. If this pipe is in the list multiple times, it will be prevented from calling this function needlessly.

The results of this change are as follows:

image

### CODE REVIEW

**Comment by Shadowmech88** [comment]
You can paste images directly into the PR body, you know.

File: code/ATMOSPHERICS/datum_pipeline.dm [hunk 1]
```diff
@@ -229,3 +232,19 @@
 			air.temperature -= heat/total_heat_capacity
 	if(network)
 		network.update = 1
+
+
+//Slows down the rate of update_icon() when many of them are called simultaneously and prevents some duplicate processing
+/datum/pipeline/proc/process_update_icon_queue()
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/datum_pipeline.dm]: you never call this.

Comment (Chewyfungus) [reply code/ATMOSPHERICS/datum_pipeline.dm]: oh man what a boof, I'll upload the fix in a few hours, thanks

**Comment by Intigracy** [comment]
"a few hours"

**Comment by Chewyfungus** [comment]
well shit
I got it all working but then the actual advantage I thought I had obtained turns out to have never existed at all
I'm gonna ask clusterfack for his opinion

**Pull request closed without merging**